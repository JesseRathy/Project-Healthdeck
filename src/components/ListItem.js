import React, { Component } from 'react';
import { Text, TouchableWithoutFeedback, View, Image } from 'react-native' ;
import { Actions } from 'react-native-router-flux'
import { CardSectionStyle } from './common';

//Retrieves a list of items for a specefic user which is specefied
// through firebase and userId
class ListItem extends Component {

//If the row of the card is pressed then we pass over the props of that card
// to the showCard file which then is rendered due to our Router.js component navigating
// to that screen through the Action method
  onRowPress() {
    Actions.showCard({ myCards: this.props.myCards });
  }

//Render every card
  render () {
    const { name, bio, pro, image } = this.props.myCards;

    return (
      <TouchableWithoutFeedback onPress={this.onRowPress.bind(this)}>
        <View>
          <CardSectionStyle>
            <View style={styles.thumbnailContainerStyle}>
              <Image
               style ={styles.thumbnailStyle}
               source={{uri: image}}
                />
            </View>
            <Text style={styles.titleStyle}>
                {pro} {": "} {name}
            </Text>
          </CardSectionStyle>
        </View>
      </TouchableWithoutFeedback>
    );
  }

}

const styles = {
  titleStyle: {
    fontSize:18,
    paddingLeft:15
  },
  thumbnailContainerStyle: {
    justifyContent: 'center',
    alignItems: 'center',
    marginLeft: 10,
    marginRight: 10
  },
  thumbnailStyle: {
    height: 50,
    width: 50
  }
}
export default ListItem;
