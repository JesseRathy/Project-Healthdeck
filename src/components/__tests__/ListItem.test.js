
import { shallow,configure,mount } from 'enzyme';
import React from 'react';
import configureStore from 'redux-mock-store';
import {connect} from 'react-redux';
import Adapter from 'enzyme-adapter-react-16';
import sinon from "sinon";
import ListItem from '../ListItem';
//Imports to get Enzyme to work if possible -> will work on later
configure({ adapter: new Adapter() });

test('renders correctly', () => {


  const wrapper = shallow(<ListItem myCards= {
    /*I've just created a const of all the values and called the card with these values.
    I have to call it */
  { name:'Test',
    bio:'I like Cats.',
    pro:"Doctor",
    image:"https://imgur.com/gallery/9vPpm4r"
  }
  }/>);
  expect(wrapper).toMatchSnapshot();
});

test('it puts each item in the list item correctly', () => {


  const wrapper = mount(<ListItem myCards= {
    /*I've just created a const of all the values and called the card with these values.
    I'm calling it like this for now as this works, but later I'd like to test and see if it works with putting them in one card and testing like that.
    --Jesse   */
  { name:'Steven',
    bio:'Fan of Matryoshka dolls and cicadas.',
    pro:"Physician",
    image:"https://imgur.com/gallery/9vPpm4r"
  }}/>);
  const myProps = wrapper.props();
  console.log(myProps);
  console.log('\n');
  const nameProp = myProps.myCards.name
  /*name Log*/
  console.log("Name =" + nameProp);

  const proProp = myProps.myCards.pro;
  /*profession log*/
  console.log("Profession =" + proProp);

  const imageProp = myProps.myCards.image;
  console.log("Image =" + imageProp);
  /*
  You have to get the props of whatever is mounted
  then you gotta call that prop even though you just got it
  then you gotta get the prop of that prop.
  Note: You have to do some 'deep diving' into each child to get the prop to compare to.
  I'll console.log() these to show it's actually pulling out what I need, and put them each in their own prop.
  T̶h̶e̶r̶e̶ ̶h̶a̶s̶ ̶t̶o̶ ̶b̶e̶ ̶a̶ ̶b̶e̶t̶t̶e̶r̶ ̶w̶a̶y̶ ̶t̶h̶e̶n̶ ̶j̶u̶s̶t̶ ̶d̶e̶e̶p̶ ̶r̶e̶a̶c̶h̶i̶n̶g̶ ̶i̶n̶t̶o̶ ̶e̶a̶c̶h̶ ̶o̶f̶ ̶t̶h̶e̶s̶e̶ ̶t̶o̶ ̶g̶e̶t̶ ̶o̶n̶e̶ ̶p̶r̶o̶p̶;̶ ̶I̶'̶l̶l̶ ̶i̶n̶v̶e̶s̶t̶i̶g̶a̶t̶e̶ ̶l̶a̶t̶e̶r̶ ̶i̶f̶ ̶I̶ ̶h̶a̶v̶e̶ ̶t̶i̶m̶e̶.
  EDIT:
  The better way is to just mount it like normal; it's a lot saner to call the props of the thing.
  Seems a little suspicious, I'll look at the snapshot later to make sure it's actually doing stuff.
  /*

  --Jesse*/

  expect(nameProp).toEqual('Steven');
  expect(proProp).toEqual("Physician");
  expect(imageProp).toEqual('https://imgur.com/gallery/9vPpm4r');

});
