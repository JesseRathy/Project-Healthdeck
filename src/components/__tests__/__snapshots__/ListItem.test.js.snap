// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ListItem
    myCards={
      Object {
        "bio": "I like Cats.",
        "image": "https://imgur.com/gallery/9vPpm4r",
        "name": "Test",
        "pro": "Doctor",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Component>
        <CardSectionStyle>
          <Component
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "marginLeft": 10,
                "marginRight": 10,
              }
            }
          >
            <Component
              source={
                Object {
                  "uri": "https://imgur.com/gallery/9vPpm4r",
                }
              }
              style={
                Object {
                  "height": 50,
                  "width": 50,
                }
              }
            />
          </Component>
          <Component
            style={
              Object {
                "fontSize": 18,
                "paddingLeft": 15,
              }
            }
          >
            Doctor
             
            : 
             
            Test
          </Component>
        </CardSectionStyle>
      </Component>,
      "onPress": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <CardSectionStyle>
          <Component
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "marginLeft": 10,
                "marginRight": 10,
              }
            }
          >
            <Component
              source={
                Object {
                  "uri": "https://imgur.com/gallery/9vPpm4r",
                }
              }
              style={
                Object {
                  "height": 50,
                  "width": 50,
                }
              }
            />
          </Component>
          <Component
            style={
              Object {
                "fontSize": 18,
                "paddingLeft": 15,
              }
            }
          >
            Doctor
             
            : 
             
            Test
          </Component>
        </CardSectionStyle>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Component
              style={
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "marginLeft": 10,
                  "marginRight": 10,
                }
              }
            >
              <Component
                source={
                  Object {
                    "uri": "https://imgur.com/gallery/9vPpm4r",
                  }
                }
                style={
                  Object {
                    "height": 50,
                    "width": 50,
                  }
                }
              />
            </Component>,
            <Component
              style={
                Object {
                  "fontSize": 18,
                  "paddingLeft": 15,
                }
              }
            >
              Doctor
               
              : 
               
              Test
            </Component>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Component
                source={
                  Object {
                    "uri": "https://imgur.com/gallery/9vPpm4r",
                  }
                }
                style={
                  Object {
                    "height": 50,
                    "width": 50,
                  }
                }
              />,
              "style": Object {
                "alignItems": "center",
                "justifyContent": "center",
                "marginLeft": 10,
                "marginRight": 10,
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "source": Object {
                  "uri": "https://imgur.com/gallery/9vPpm4r",
                },
                "style": Object {
                  "height": 50,
                  "width": 50,
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                "Doctor",
                " ",
                ": ",
                " ",
                "Test",
              ],
              "style": Object {
                "fontSize": 18,
                "paddingLeft": 15,
              },
            },
            "ref": null,
            "rendered": Array [
              "Doctor",
              " ",
              ": ",
              " ",
              "Test",
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Component>
          <CardSectionStyle>
            <Component
              style={
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "marginLeft": 10,
                  "marginRight": 10,
                }
              }
            >
              <Component
                source={
                  Object {
                    "uri": "https://imgur.com/gallery/9vPpm4r",
                  }
                }
                style={
                  Object {
                    "height": 50,
                    "width": 50,
                  }
                }
              />
            </Component>
            <Component
              style={
                Object {
                  "fontSize": 18,
                  "paddingLeft": 15,
                }
              }
            >
              Doctor
               
              : 
               
              Test
            </Component>
          </CardSectionStyle>
        </Component>,
        "onPress": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <CardSectionStyle>
            <Component
              style={
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "marginLeft": 10,
                  "marginRight": 10,
                }
              }
            >
              <Component
                source={
                  Object {
                    "uri": "https://imgur.com/gallery/9vPpm4r",
                  }
                }
                style={
                  Object {
                    "height": 50,
                    "width": 50,
                  }
                }
              />
            </Component>
            <Component
              style={
                Object {
                  "fontSize": 18,
                  "paddingLeft": 15,
                }
              }
            >
              Doctor
               
              : 
               
              Test
            </Component>
          </CardSectionStyle>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Component
                style={
                  Object {
                    "alignItems": "center",
                    "justifyContent": "center",
                    "marginLeft": 10,
                    "marginRight": 10,
                  }
                }
              >
                <Component
                  source={
                    Object {
                      "uri": "https://imgur.com/gallery/9vPpm4r",
                    }
                  }
                  style={
                    Object {
                      "height": 50,
                      "width": 50,
                    }
                  }
                />
              </Component>,
              <Component
                style={
                  Object {
                    "fontSize": 18,
                    "paddingLeft": 15,
                  }
                }
              >
                Doctor
                 
                : 
                 
                Test
              </Component>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Component
                  source={
                    Object {
                      "uri": "https://imgur.com/gallery/9vPpm4r",
                    }
                  }
                  style={
                    Object {
                      "height": 50,
                      "width": 50,
                    }
                  }
                />,
                "style": Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "marginLeft": 10,
                  "marginRight": 10,
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "source": Object {
                    "uri": "https://imgur.com/gallery/9vPpm4r",
                  },
                  "style": Object {
                    "height": 50,
                    "width": 50,
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  "Doctor",
                  " ",
                  ": ",
                  " ",
                  "Test",
                ],
                "style": Object {
                  "fontSize": 18,
                  "paddingLeft": 15,
                },
              },
              "ref": null,
              "rendered": Array [
                "Doctor",
                " ",
                ": ",
                " ",
                "Test",
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
